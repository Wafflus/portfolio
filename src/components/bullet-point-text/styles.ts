import styled from "styled-components";

export const BulletPointTextStyled = styled.li`
    color: ${({ theme }) => theme.data.text.lightColor};
    font-family: '${({ theme }) => theme.data.fontFamilies.outfit}', sans-serif;
    font-size: 1.125rem;
    letter-spacing: 0.1rem;
    line-height: 120%;
    list-style-type: none;
    position: relative;
    text-align: center;
    transition: transform 0.5s;

    &:hover {
        transform: scale(102%);
    }
    
    ${props => props.theme.breakpoint.above('lg',
        `
            &::before {
                background-color: ${props.theme.data.secondaryColor};
                border-radius: 50%;
                content: '';
                height: 20px;
                left: -40px;
                position: absolute;
                transition: background-color 0.5s;
                width: 20px;
            }

            &:hover::before {
                background-color: ${props.theme.data.primaryColor};
            }
        `
    )}
`;