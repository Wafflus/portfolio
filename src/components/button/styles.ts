import { IButtonStyledProps } from "./types";
import styled, { css } from "styled-components";

export const ButtonStyled = styled.a<IButtonStyledProps>`
    align-items: center;
    background-color: ${({ theme }) => theme.data.secondaryColor};
    border-radius: 8px;
    color: ${({ theme }) => theme.data.text.darkColor};
    display: flex;
    font-size: 2rem;
    font-variant: small-caps;
    gap: 16px;
    height: 66px;
    justify-content: center;
    letter-spacing: 0.1rem;
    max-width: 300px;
    min-width: 200px;
    padding: 8px;
    text-decoration: none;
    transition: transform 0.5s;
    width: 100%;

    ${props => props.outline && css`
        background-color: transparent;
        border: 1px solid ${({ theme }) => theme.data.primaryColor};
        color: #93855B;
    `}

    &:hover {
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
        transform: scale(102%);
    }
`;