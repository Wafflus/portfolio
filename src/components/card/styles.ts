import styled from "styled-components";

export const CardStyled = styled.article`
    border-radius: 8px;
    padding: 16px;

    &:nth-child(odd) {
        background-color: ${({ theme }) => theme.data.highlightColor};
    }

    &:hover {
        outline: 3px solid ${({ theme }) => theme.data.borderColor};

        & > div > img {
            border: 3px solid ${({ theme }) => theme.data.primaryColor};
            transform: scale(102%);
        }
    }

    ${props => props.theme.breakpoint.above('lg',
        `
            &:nth-child(even) > div {
                flex-direction: row-reverse;
            }
        `
    )}
`;

export const TitleStyled = styled.h3`
    color: ${({ theme }) => theme.data.text.darkColor};
    font-size: 1.5rem;
    font-variant: small-caps;
    font-weight: 600;
    letter-spacing: 0.1rem;
    text-align: center;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            font-size: 2rem;
            text-align: start;
        `
    )}
`;

export const ToolsStyled = styled.h4`
    color: ${({ theme }) => theme.data.text.defaultColor};
    font-variant: small-caps;
    letter-spacing: 0.1rem;
    text-align: center;
`;

export const DescriptionStyled = styled.p`
    color: ${({ theme }) => theme.data.text.defaultColor};
    flex: 1;
    letter-spacing: 0.1rem;
`;