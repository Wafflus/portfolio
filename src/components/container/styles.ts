import { IContainerStyledProps } from "./types";
import styled from "styled-components";

export const ContainerStyled = styled.div<IContainerStyledProps>`
    height: 100%;
    max-width: 1440px;
    padding: ${props => props.verticalPadding} ${props => props.horizontalPadding};
    width: 100%;
    z-index: 200;
`;