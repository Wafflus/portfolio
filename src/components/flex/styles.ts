import { IFlexStyledProps } from "./types";
import styled from "styled-components";

export const FlexStyled = styled.div<IFlexStyledProps>`
    align-items: ${props => props.defaultFlexProps?.alignItems};
    display: flex;
    flex-direction: ${props => props.defaultFlexProps?.flexDirection};
    gap: ${props => props.defaultFlexProps?.gap};
    justify-content: ${props => props.defaultFlexProps?.justifyContent};
    max-width: ${props => props.defaultFlexProps?.maxWidth};
    width: 100%;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            align-items: ${props.largeBreakpointFlexProps?.alignItems};
            flex-direction: ${props.largeBreakpointFlexProps?.flexDirection};
            gap: ${props.largeBreakpointFlexProps?.gap};
            justify-content: ${props.largeBreakpointFlexProps?.justifyContent};
            max-width: ${props.largeBreakpointFlexProps?.maxWidth};
        `
    )}
`;