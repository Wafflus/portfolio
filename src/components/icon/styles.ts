import styled from "styled-components";

import { IImageProps } from "./types";

export const IconStyled = styled.img<IImageProps>`
    width: ${props => props.size}${props => props.unit};
    height: ${props => props.size}${props => props.unit};
    border-radius: 50%;
`;