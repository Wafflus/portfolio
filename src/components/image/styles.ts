import styled from "styled-components";

export const ImageStyled = styled.img`
    border-radius: 8px;
    border: 3px solid transparent;
    height: 100%;
    max-height: 350px;
    max-width: 500px;
    transition: transform 0.5s, border 0.5s;
    width: 100%;
`;