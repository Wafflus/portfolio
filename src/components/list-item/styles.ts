import { IListItemStyledProps } from "./types";
import styled from "styled-components";

export const ListItemStyled = styled.li<IListItemStyledProps>`
    list-style-type: ${props => props.listStyleType};
    position: relative;
`;