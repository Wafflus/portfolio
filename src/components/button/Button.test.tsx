import "jest-styled-components";
import { Button } from ".";
import { render } from "@testing-library/react";
import { theme } from "styles";
import { ThemeProvider } from "styled-components";

test('button renders', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Button href="https://example.com">
                <>Button</>
            </Button>
        </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
});

test('should render without outline by default', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Button href="https://example.com">
                <>Button</>
            </Button>
        </ThemeProvider>
    );

    expect(container.firstChild).not.toHaveStyleRule('background-color', 'transparent');
    expect(container.firstChild).not.toHaveStyleRule('border', `1px solid ${theme.data.primaryColor}`);
    expect(container.firstChild).not.toHaveStyleRule('color', '#93855B');
});

test('should render with outline when outline prop is true', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Button href="https://example.com" buttonStylesProps={{ outline: true }}>
                <>Button</>
            </Button>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('background-color', 'transparent');
    expect(container.firstChild).toHaveStyleRule('border', `1px solid ${theme.data.primaryColor}`);
    expect(container.firstChild).toHaveStyleRule('color', '#93855B');
});

test('should open link in a new tab', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Button href="https://example.com">
                <>Button</>
            </Button>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveAttribute('target', '_blank');
});

test('should have box shadow on hover', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Button href="https://example.com">
                <>Button</>
            </Button>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('box-shadow', '0px 4px 4px rgba(0,0,0,0.1)', {
        modifier: ':hover'
    });
});

test('should scale on hover', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Button href="https://example.com">
                <>Button</>
            </Button>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('transform', 'scale(102%)', {
        modifier: ':hover'
    });
});

test('should be a link of type button', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Button href="https://example.com">
                <>Button</>
            </Button>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveAttribute('type', 'button');
});