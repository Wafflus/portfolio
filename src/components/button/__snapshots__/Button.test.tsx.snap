// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFECB1;
  border-radius: 8px;
  color: #756A47;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 2rem;
  font-variant: small-caps;
  gap: 16px;
  height: 66px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  max-width: 300px;
  min-width: 200px;
  padding: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: -webkit-transform 0.5s;
  -webkit-transition: transform 0.5s;
  transition: transform 0.5s;
  width: 100%;
}

.c0:hover {
  box-shadow: 0px 4px 4px rgba(0,0,0,0.1);
  -webkit-transform: scale(102%);
  -ms-transform: scale(102%);
  transform: scale(102%);
}

<a
  class="c0"
  href="https://example.com"
  target="_blank"
  type="button"
>
  Button
</a>
`;
